### PR TITLE
chore: update error when not-in-okteto-context for guiding next step more clearly (fix #3792)

### LINKED ISSUE #3792: okteto CLI doesn't recognize a cluster with okteto installed if you pick one of the clusters added via kubeconfig
**Describe the bug**
I have a GKE cluster where I installed Okteto. Since the cluster is present in my `kubeconfig` it starts showing in the list when I do `okteto ctx use`. But when I select it and try to deploy a repository, it says:

```
 x  'build.api-dev.image' is required if your cluster doesn't have Okteto installed
```
Which isn't correct. It works when I add the context by specifying the Okteto URL.

** Steps to replicate**
- Create a cluster and add it to your kubeconfig
- Install Okteto in it
- Choose the cluster from `okteto ctx use` 
- Run `okteto deploy` or `okteto up` from the movies app
It will fail with the message that the cluster doesn't have Okteto installed.
- Add the cluster via the URL for the Okteto installation (for example: `https://demo.okteto.com`)
- Run `okteto deploy` or `okteto up` from the movies app.

It will now work even though the cluster you have chosen is technically the same one. 

**Expected behavior**
Ideally, it should work when I choose the cluster from the `okteto ctx use` list without even adding via the URL. But if that's not possible maybe we can update the error message to say to add the cluster to the context via the URL instead of saying that the cluster doesn't have Okteto installed, which can be confusing.

### PR DESCRIPTION
**Pull request by minhoryang**
# Proposed changes

Fixes #3792

## How to validate

1. `okteto context` and choose a pure kubernetes cluster (whether Okteto installed or not)
1. `git clone https://github.com/okteto/movies-with-compose && cd movies-with-compose`
1. `okteto up -l debug` and one of these error shows:
> `'build.frontend.image' is required if your cluster doesn't have Okteto installed`
> `'build.api.image' is required if your cluster doesn't have Okteto installed`
> `'build.movies.image' is required if your cluster doesn't have Okteto installed`
> `'build.init.image' is required if your cluster doesn't have Okteto installed`

## CLI Quality Reminders 🔧

For both authors and reviewers:

- Scrutinize for potential regressions
- Ensure key automated tests are in place
- Build the CLI and test using the validation steps
- Assess Developer Experience impact (log messages, performances, etc)
- If too broad, consider breaking into smaller PRs
- Adhere to our [code style](https://github.com/okteto/okteto/blob/master/docs/code-style.md) and [code review](https://github.com/okteto/okteto/blob/master/docs/code-review.md) guidelines

- [x] ran `make lint`

### CODE REVIEW

**Review by jLopezbarb — APPROVED** [review]

File: cmd/build/v2/build.go [hunk 1]
```diff
@@ -191,7 +191,7 @@ func (bc *OktetoBuilder) Build(ctx context.Context, options *types.BuildOptions)
 			}
 
 			if !okteto.Context().IsOkteto && buildSvcInfo.Image == "" {
-				return fmt.Errorf("'build.%s.image' is required if your cluster doesn't have Okteto installed", svcToBuild)
+				return fmt.Errorf("'build.%s.image' is required if your context doesn't have Okteto installed", svcToBuild)
```
Comment (AdrianPedriza) [inline cmd/build/v2/build.go:194]: I am not very sure about this change. The current `` message is the one used at various points when trying to perform a certain type of operation on a cluster that does not have okteto installed. If we change this here we should change it at the other points, for consistency. @jLopezbarb

Comment (jLopezbarb) [reply cmd/build/v2/build.go:194]: This was a decission that came from the marketing team given that there were issues understanding what is a cluster vs a context. I agree that we should change all other messages from cluster to context. @minhoryang Are you ok with doing those changes?

Comment (minhoryang) [reply cmd/build/v2/build.go:194]: @jLopezbarb @AdrianPedriza Yes, it's okay to replace all `cluster`-mentioning messages. but need to grab these sentences first. I'll try to dive in.

**Review by andreafalzetti — APPROVED** [review]

File: cmd/build/v2/build.go (hunk 1 above)
Comment (minhoryang) [reply cmd/build/v2/build.go:194]: Changing `cluster` to `context`:
- `pkg/errors/errors.go`: (I think) not okay to change.
  - [ ] ErrNotOktetoCluster
  - [ ] ErrDivertNotSupported
  - [ ] ErrContextIsNotOktetoCluster
- `cmd/kubeconfig.go`: (I think) not okay to change.
- `cmd/build/v2/build.go`: okay to change.
- `cmd/context/context.go`: [partially](https://github.com/okteto/okteto/blob/master/cmd/context/context.go#L67) okay to change. it explains [the difference between `context` and `cluster`](https://github.com/okteto/okteto/blob/master/cmd/context/context.go#L27-L40).
- `cmd/context/create.go`: (I think) not okay to change. it explains [the difference between `context` and `cluster`](https://github.com/okteto/okteto/blob/master/cmd/context/create.go#L89-L110)
- `cmd/context/update-kubeconfig.go`: (I think) not okay to change.
- ...

Changing `okteto cluster` to `okteto context` might be good.

Comment (minhoryang) [reply cmd/build/v2/build.go:194]: I made a simple commit for reviewing: https://github.com/okteto/okteto/pull/4011/commits/08b03021773f4e1ec79fb6bc12884e5e93bde39b

**Comment by andreafalzetti** [comment]
Moving to draft because lacking testing and more changes.

**Comment by andreafalzetti** [comment]
@minhoryang could you please resolve the conflicts?

File: cmd/context/update-kubeconfig.go [hunk 2]
```diff
@@ -57,7 +57,7 @@ func UpdateKubeconfigCMD(okClientProvider oktetoClientProvider) *cobra.Command {
 		Hidden: true,
 		Use:    "update-kubeconfig",
 		Args:   utils.NoArgsAccepted("https://okteto.com/docs/reference/cli/#kubeconfig"),
-		Short:  "Download credentials for the Kubernetes cluster selected via 'okteto context'",
+		Short:  "Download credentials for the Kubernetes context selected via 'okteto context'",
```
Comment (andreafalzetti) [inline cmd/context/update-kubeconfig.go:60]: from a product perspective @ifbyol do you think it's a good idea to also say "Kubernetes context"?

Comment (ifbyol) [reply cmd/context/update-kubeconfig.go:60]: I'm not sure, with `okteto context` you don't select a `Kubernetes context`, you select an `Okteto context`, which is connected to a Kubernetes cluster. So, I wouldn't change it

Comment (andreafalzetti) [reply cmd/context/update-kubeconfig.go:60]: It's also odd to me changing this. @minhoryang can you please revert the changes in reference to kubernetes explicitly?

File: cmd/build/v2/build.go (hunk 1 above)
Comment (AdrianPedriza) [inline cmd/build/v2/build.go:194]:
```suggestion
				return fmt.Errorf("'build.%s.image' is required if your context does not belong to Okteto.", svcToBuild)
```
I think the term `install` was focused on the `cluster`. Now that we talk about `context` I think it is more appropriate in this way.

File: cmd/build/v2/build.go [hunk 3]
```diff
@@ -223,8 +223,8 @@ func (bc *OktetoBuilder) buildService(ctx context.Context, manifest *model.Manif
 	switch {
 	case shouldAddVolumeMounts(buildSvcInfo) && !okteto.IsOkteto():
 		return "", oktetoErrors.UserError{
-			E:    fmt.Errorf("Build with volume mounts is not supported on vanilla clusters"),
-			Hint: "Please connect to a okteto cluster and try again",
+			E:    fmt.Errorf("Build with volume mounts is not supported on vanilla contexts"),
```
Comment (AdrianPedriza) [inline cmd/build/v2/build.go:226]:
```suggestion
			E:    fmt.Errorf("Build with volume mounts is not supported in non-Okteto contexts."),
```

File: cmd/up/activate.go [hunk 4]
```diff
@@ -194,7 +194,7 @@ func (up *upContext) activate() error {
 			if isWatchesConfigurationTooLow(watches) {
 				folder := config.GetNamespaceHome(up.Dev.Namespace)
 				if utils.GetWarningState(folder, ".remotewatcher") == "" {
-					oktetoLog.Yellow("The value of /proc/sys/fs/inotify/max_user_watches in your cluster nodes is too low.")
+					oktetoLog.Yellow("The value of /proc/sys/fs/inotify/max_user_watches in your context nodes is too low.")
```
Comment (AdrianPedriza) [inline cmd/up/activate.go:197]:
```suggestion
					oktetoLog.Yellow("The value of /proc/sys/fs/inotify/max_user_watches in your cluster nodes is too low.")
```
I thing in this case is clearer talk about cluster

File: cmd/up/activate.go [hunk 5]
```diff
@@ -382,7 +382,7 @@ func (up *upContext) waitUntilDevelopmentContainerIsRunning(ctx context.Context,
 			// this provides 2 min for "FailedScheduling" to resolve by themselves
 			if strings.Contains(failedSchedulingEvent.Message, "Insufficient cpu") || strings.Contains(failedSchedulingEvent.Message, "Insufficient memory") {
 				return oktetoErrors.UserError{E: fmt.Errorf("insufficient resources"),
-					Hint: "Increase cluster resources or timeout of resources. More information is available here: https://okteto.com/docs/reference/manifest/#timeout-time-optional"}
+					Hint: "Increase context resources or timeout of resources. More information is available here: https://okteto.com/docs/reference/manifest/#timeout-time-optional"}
```
Comment (AdrianPedriza) [inline cmd/up/activate.go:385]:
```suggestion
					Hint: "Increase cluster resources or timeout of resources. More information is available here: https://okteto.com/docs/reference/manifest/#timeout-time-optional"}
```
I thing in this case is clearer talk about cluster

File: cmd/up/activate.go [hunk 6]
```diff
@@ -423,7 +423,7 @@ func (up *upContext) waitUntilDevelopmentContainerIsRunning(ctx context.Context,
 				continue
 			case "TriggeredScaleUp":
 				failedSchedulingEvent = nil
-				oktetoLog.Spinner("Waiting for the cluster to scale up...")
+				oktetoLog.Spinner("Waiting for the context to scale up...")
```
Comment (AdrianPedriza) [inline cmd/up/activate.go:426]:
```suggestion
				oktetoLog.Spinner("Waiting for the cluster to scale up...")
```
I thing in this case is clearer talk about cluster

File: cmd/up/up.go [hunk 7]
```diff
@@ -61,7 +61,7 @@ import (
 
 // ReconnectingMessage is the message shown when we are trying to reconnect
 const (
-	ReconnectingMessage = "Trying to reconnect to your cluster. File synchronization will automatically resume when the connection improves."
+	ReconnectingMessage = "Trying to reconnect to your context. File synchronization will automatically resume when the connection improves."
```
Comment (AdrianPedriza) [inline cmd/up/up.go:64]:
```suggestion
	ReconnectingMessage = "Trying to reconnect to your cluster. File synchronization will automatically resume when the connection improves."
```
I thing in this case is clearer talk about cluster

File: cmd/manifest/init-v1.go [hunk 8]
```diff
@@ -141,7 +141,7 @@ func (*ManifestCommand) RunInitV1(ctx context.Context, opts *InitOpts) error {
 		}
 
 		if !supportsPersistentVolumes(ctx) {
-			oktetoLog.Yellow("Default storage class not found in your cluster. Persistent volumes not enabled in your okteto manifest")
+			oktetoLog.Yellow("Default storage class not found in your context. Persistent volumes not enabled in your okteto manifest")
```
Comment (AdrianPedriza) [inline cmd/manifest/init-v1.go:144]:
```suggestion
			oktetoLog.Yellow("Default storage class not found in your cluster. Persistent volumes not enabled in your okteto manifest")
```
I thing in this case is clearer talk about cluster

**Comment by andreafalzetti** [comment]
Moved to draft until @minhoryang has time to address the feedback.

**Review by AdrianPedriza — APPROVED** [review]

**Comment by minhoryang** [comment]
@andreafalzetti @AdrianPedriza @jLopezbarb Thanks for reviewing my PR.

**Pull request merged**